### PR TITLE
Add bare IANA Suffix lint

### PR DESCRIPTION
**Pull request by kumarde**

### CODE REVIEW

File: lints/lint_dnsname_contains_bare_iana_suffix.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
+// lint_dnsname_contains_bare_iana_suffix.go
```
Comment (dadrian) [inline lints/lint_dnsname_contains_bare_iana_suffix.go]: Cut this out.

And put in the license header.

File: lints/lint_dnsname_contains_bare_iana_suffix.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	return nil
+}
+
+func (l *dnsNameContainsBareIANASuffix) CheckApplies(c *x509.Certificate) bool {
+	return util.IsSubscriberCert(c) && util.DNSNamesExist(c)
+}
+
+func (l *dnsNameContainsBareIANASuffix) Execute(c *x509.Certificate) *LintResult {
+	if c.Subject.CommonName != "" {
+		if util.IsInTLDMap(c.Subject.CommonName) {
```
Comment (dadrian) [inline lints/lint_dnsname_contains_bare_iana_suffix.go]: Why do we use the map instead of the parse function in `util`

Comment (kumarde) [reply lints/lint_dnsname_contains_bare_iana_suffix.go]: I thought you wanted IANA suffix not ICANN Public Suffix? 

The `util.ICANNPublicSuffixParse` is just the public part of the public suffix

File: lints/lint_dnsname_contains_bare_iana_suffix.go (hunk 1 above)
Comment (kumarde) [reply lints/lint_dnsname_contains_bare_iana_suffix.go]: 👍

File: lints/lint_dnsname_contains_bare_iana_suffix.go (hunk 2 above)
Comment (dadrian) [reply lints/lint_dnsname_contains_bare_iana_suffix.go]: OK, but still let @zakird do a pass

**Comment by robstradling** [comment]
I think this lint is checking the PRIVATE section of the PSL:
https://crt.sh/?zlint=908&iCAID=16418

Shouldn't it only check the ICANN section?

**Comment by kumarde** [comment]
Ah, that's not this lint, but the lint: `w_san_iana_pub_suffix_empty`. I think this doesn't use the correct `publicsuffix` package (as per other recent updates). I'll update this.

**Comment by robstradling** [comment]
@kumarde Thanks!  With so many similar lints, it's easy to get them confused.  ;-)

**Pull request merged**